Passing variable with name stderr into closure causes c compilation error

### Describe the bug

Passing a variable with the names `stdin`,`stdout`, or `stderr` into a closure causes c compilation error.

### Reproduction Steps

This should reproduce the bug:

    fn other() {
    	mut stderr:= 0
    	causes_error := fn [mut stderr] (){
    		stderr = 1
    	}
    	causes_error()
    }

### Expected Behavior

when `causes_error` is called it should do nothing but change the value of `stderr` to 1

### Current Behavior
with `v .`

    ================== C compilation error (from tcc): ==============
    cc: /tmp/v_1000/vlangbugreport.01J8729DJ5JMS8XZVEQVX6YZ5Q.tmp.c:2386: error: field not found: __v_stderr
    =================================================================
    (You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
    builder error:
    ==================
    C error found. It should never happen, when compiling pure V code.
    This is a V compiler bug, please report it using `v bug file.v`,
    or goto https://github.com/vlang/v/issues/new/choose .
    You can also use #help on Discord: https://discord.gg/vlang .

with `-cg` flag:

    /tmp/v_1000/vlangbugreport.01J86RRKF3306VGEYC560WVM8V.tmp.c:3396: error: field not found: __v_stderr
    builder error:

### Possible Solution

Seems to be related to any variable with the name `stdin`, `stdout`, or `stderr` being passed into a closure.
Further investigation reveals that the closure's context struct is exported with the original variable name as field,
but then this generates a 'sanitized' name("\_\_v_" + varname) to prevent reserved name colision. Somehow the var name
must be also be 'sanitized' when output elsewhere. (see Additional Info/Context)

### Additional Information/Context
This is the generated c file(with another non error causing function; same code different variable names). There seems to be some issue with mangling the variable name into `__v_stderr` (to prevent a conflict?) but not doing that in the struct field.

    struct _V_anon_fn_143c1662f75e5a6c__76_Ctx {
    	int stderr;
    };

    VV_LOCAL_SYMBOL void anon_fn_143c1662f75e5a6c__76(void) {
    	struct _V_anon_fn_143c1662f75e5a6c__76_Ctx* _V_closure_ctx = __CLOSURE_GET_DATA();
    	_V_closure_ctx->__v_stderr = 0;
    }


    /*F*/
    struct _V_anon_fn_143c1662f75e5a6c__182_Ctx {
    	int no_conflict;
    };

    VV_LOCAL_SYMBOL void anon_fn_143c1662f75e5a6c__182(void) {
    	struct _V_anon_fn_143c1662f75e5a6c__182_Ctx* _V_closure_ctx = __CLOSURE_GET_DATA();
    	_V_closure_ctx->no_conflict = 0;
    }


### V version

V 0.4.7 9a0f4d8

### Environment details (OS name and version, etc.)

    tungsten@Federation:~/Documents/vlangbugreport$ v doctor
    V full version: V 0.4.7 63957f2.9a0f4d8
    OS: linux, Linux version 6.9.12-100.fc39.x86_64 (mockbuild@9543a464e37f4db8bd0fde43d10019f7) (gcc (GCC) 13.3.1 20240522 (Red Hat 13.3.1-1), GNU ld version 2.40-14.fc39) #1 SMP PREEMPT_DYNAMIC Sat Jul 27 16:09:11 UTC 2024
    Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 5800X 8-Core Processor

    getwd: /home/tungsten/Documents/vlangbugreport
    vexe: /home/tungsten/v/v
    vexe mtime: 2024-09-20 03:16:33

    vroot: OK, value: /home/tungsten/v
    VMODULES: OK, value: /home/tungsten/.vmodules
    VTMP: OK, value: /tmp/v_1000

    Git version: git version 2.45.2
    Git vroot status: weekly.2024.37-30-g9a0f4d84
    .git/config present: true

    CC version: cc (GCC) 13.3.1 20240522 (Red Hat 13.3.1-1)
    thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -592,6 +592,7 @@ fn (mut g Gen) gen_anon_fn(mut node ast.AnonFn) {
 	for var in node.inherited_vars {
 		mut has_inherited := false
 		mut is_ptr := false
+		var_name := c_name(var.name)
 		if obj := node.decl.scope.find(var.name) {
 			if obj is ast.Var {
 				is_ptr = obj.typ.is_ptr()
@@ -599,13 +600,13 @@ fn (mut g Gen) gen_anon_fn(mut node ast.AnonFn) {
 					has_inherited = true
 					var_sym := g.table.sym(var.typ)
 					if var_sym.info is ast.ArrayFixed {
-						g.write('.${var.name} = {')
+						g.write('.${var_name} = {')
 						for i in 0 .. var_sym.info.size {
-							g.write('${closure_ctx}->${var.name}[${i}],')
+							g.write('${closure_ctx}->${var_name}[${i}],')
 						}
 						g.writeln('},')
 					} else {
-						g.writeln('.${var.name} = ${closure_ctx}->${var.name},')
+						g.writeln('.${var_name} = ${closure_ctx}->${var_name},')
 					}
 				}
 			}
@@ -613,15 +614,15 @@ fn (mut g Gen) gen_anon_fn(mut node ast.AnonFn) {
 		if !has_inherited {
 			var_sym := g.table.sym(var.typ)
 			if var_sym.info is ast.ArrayFixed {
-				g.write('.${var.name} = {')
+				g.write('.${var_name} = {')
 				for i in 0 .. var_sym.info.size {
-					g.write('${var.name}[${i}],')
+					g.write('${var_name}[${i}],')
 				}
 				g.writeln('},')
 			} else if g.is_autofree && !var.is_mut && var_sym.info is ast.Array {
-				g.writeln('.${var.name} = array_clone(&${var.name}),')
+				g.writeln('.${var_name} = array_clone(&${var_name}),')
 			} else if g.is_autofree && !var.is_mut && var_sym.kind == .string {
-				g.writeln('.${var.name} = string_clone(${var.name}),')
+				g.writeln('.${var_name} = string_clone(${var_name}),')
 			} else {
 				mut is_auto_heap := false
 				if obj := node.decl.scope.parent.find(var.name) {
@@ -630,9 +631,9 @@ fn (mut g Gen) gen_anon_fn(mut node ast.AnonFn) {
 					}
 				}
 				if is_auto_heap && !is_ptr {
-					g.writeln('.${var.name} = *${var.name},')
+					g.writeln('.${var_name} = *${var_name},')
 				} else {
-					g.writeln('.${var.name} = ${var.name},')
+					g.writeln('.${var_name} = ${var_name},')
 				}
 			}
 		}
@@ -661,11 +662,11 @@ fn (mut g Gen) gen_anon_fn_decl(mut node ast.AnonFn) {
 			var_sym := g.table.sym(var.typ)
 			if var_sym.info is ast.FnType {
 				sig := g.fn_var_signature(var_sym.info.func.return_type, var_sym.info.func.params.map(it.typ),
-					var.name)
+					c_name(var.name))
 				builder.writeln('\t' + sig + ';')
 			} else {
 				styp := g.typ(var.typ)
-				builder.writeln('\t${styp} ${var.name};')
+				builder.writeln('\t${styp} ${c_name(var.name)};')
 			}
 		}
 		builder.writeln('};\n')
@@ -2177,7 +2178,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 								mut is_ptr := false
 								if i == 0 {
 									if obj.is_inherited {
-										g.write(closure_ctx + '->' + node.name)
+										g.write(closure_ctx + '->' + c_name(node.name))
 									} else {
 										g.write(node.name)
 									}
@@ -2199,7 +2200,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 							}
 							is_fn_var = true
 						} else if obj.is_inherited {
-							g.write(closure_ctx + '->' + node.name)
+							g.write(closure_ctx + '->' + c_name(node.name))
 							is_fn_var = true
 						}
 					}

--- a/vlib/v/tests/fns/anon_c_keywords_closure_test.v
+++ b/vlib/v/tests/fns/anon_c_keywords_closure_test.v
@@ -1,0 +1,18 @@
+fn test_main() {
+	mut stderr := 0
+	mut stdout := 0
+	mut stdin := fn () {}
+
+	causes_error := fn [mut stderr, mut stdin, mut stdout] () int {
+		stderr = 0
+		stdin()
+		stdout = 2
+		assert stderr == 0
+		assert stdout == 2
+		return stderr + stdout
+	}
+	assert causes_error() == 2
+	assert voidptr(C.stdin) != unsafe { nil }
+	assert voidptr(C.stdout) != unsafe { nil }
+	assert voidptr(C.stderr) != unsafe { nil }
+}

